org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/binding/BindingsView.java

```diff
@@ -252,7 +262,51 @@ public void connectionRemoved(IConnectedEngineFacade server) {
     for (var server : SonarLintCorePlugin.getServersManager().getServers()) {
       server.addConnectedEngineListener(serverListener);
     }
-
+    
+    // add listener for when project is opened / closed / deleted
+    projectListener = event -> {
+      if (event.getType() == IResourceChangeEvent.PRE_CLOSE
+        || event.getType() == IResourceChangeEvent.PRE_DELETE) {
+        shouldRefresh = true;
+      } else if (event.getDelta() != null) {
+        try {
+          event.getDelta().accept((IResourceDelta delta) -> {
+            if (delta.getKind() == IResourceDelta.CHANGED) {
+              // INFO: With adding "IResourceDelta.ADDED" to the mask we can also update the binding view when
+              //       importing a project. Therefore the user can see if a newly imported project is already
+              //       correctly binded.
+              for (var child : delta.getAffectedChildren(IResourceDelta.CHANGED)) {
```

Comment:
FYI: The delta resource is the Workspace and therefore we have to check if the direct children have changed. We specifically only look for changed projects (excluding hidden ones like "External Plug-in Libraries").